app/models/folio/bib_data.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+module Folio
+  # Bibliographic data from Folio.
+  class BibData
+    def initialize(request, _ = nil)
+      @request = request
+    end
+
+    attr_reader :request
+
+    def title
+      json.fetch('indexTitle', '')
+    end
+
+    def author
+      json.fetch('contributors', []).map { |contrib| contrib.fetch('name') }.join('; ')
+    end
+
+    def pub_date
+      json.fetch('publication', []).pick('dateOfPublication')
+    end
+
+    def format
+      type_id = json.fetch('instanceTypeId')
+      return unless type_id
+
+      type_labels = folio_client.instance_types.each_with_object({}) { |type, result| result[type['id']] = type['name'] }
+
+      type_labels[type_id]
+    end
+
+    def isbn
+      json.fetch('identifiers', []).filter_map do |identifier|
+        identifier.fetch('value') if identifier.fetch('identifierTypeId') == Settings.folio.isbn_identifier_type
+      end.join('; ')
+    end
+
+    def finding_aid
+      json.fetch('electronicAccess', []).find do |access|
+        access.fetch('materialsSpecification') == 'Finding aid available online'
+      end&.fetch('uri')
+    end
+
+    def holdings
+      json['holdings']
+    end
+
+    def finding_aid?
+      finding_aid.present?
+    end
+
+    def view_url
+      [Settings.searchworks_api, 'view', request.item_id].join('/')
+    end
+
+    def instance_id
+      @instance_id ||= folio_client.resolve_to_instance_id(hrid: "a#{request.item_id}")
```

Comment:
I think we only want to append the `a` with a numeric hrid for backwards compatibility (and nicer looking URLs). We'll have other hrids with other prefixes (e.g. `L0123`) that shouldn't have that prefix.

Reply:
But the request coming from searchworks, doesn't use the `a` prefix.  So I'm not sure how we expect to find the record in Folio if we don't append it at some point.  I'm open to suggestions.

Reply:
Like I said, I think we need to check if it's a numeric catkey. If it is, toss on the `a` prefix, but if it already has a letter, don't.

Reply:
Okay, I don't think I understood what you were saying originally.

Reply:
Fixed now.